service/insights_v3.go

```diff
@@ -0,0 +1,24 @@
+package service
+
+import (
+	. "github.com/talbright/go-desk/types"
+	. "github.com/talbright/go-desk/resource"
+	"net/http"
+)
+
+type InsightsV3Service struct {
+	client *Client
+}
+
+func (s *InsightsV3Service) Report(query *InsightsV3Query) (*InsightsV3Report, *http.Response, error) {
```

Comment:
As long as a 429 response is returned from the API then the changes in #13 should be able to handle it.. but yes an integration test for this is a great idea.

Looks like `X-Rate-Limit-Remaining` is passed along with these requests. Based on #13, if a 429 is returned then the client would sleep until `X-Rate-Limit-Reset` has passed and then the request would be retried. Assuming no other insights requests are being made then the subsequent request should succeed.